### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by gshennvm**
# What does this PR do ?

upgrade dockerfile and package versions to latest nemo release

### CODE REVIEW

File: nemo_aligner/package_info.py [hunk 1]
```diff
@@ -14,9 +14,9 @@
 
 
 MAJOR = 0
-MINOR = 1
+MINOR = 2
 PATCH = 0
-PRE_RELEASE = "rc0"
+PRE_RELEASE = ""
```
Comment (odelalleau) [inline nemo_aligner/package_info.py:19]: Should we use something like `0.3.0.dev` since technically we are now working on the next version in `main`?

**Review by odelalleau — APPROVED** [review]
Looks good to me, I assume the new Dockerfile matches the one from the new NeMo.

File: nemo_aligner/package_info.py (hunk 1 above)
Comment (gshennvm) [reply nemo_aligner/package_info.py]: yup! fixed https://github.com/NVIDIA/NeMo-Aligner/commit/62005260051edb5d9ea4794fa1fc3b170155e239

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ## [Next Version]
 
+### New features and optimizations
+
+### Breaking changes
+
+### Bug Fixes
+- Fixed issue where random sampler keeps state when resetting for validation, leading to a different validation batch each validation step. Fixed by using a deterministic sampler
```
Comment (gshennvm) [inline CHANGELOG.md:13]: @odelalleau also fixed this for comment with the random sampler fixes

Comment (odelalleau) [reply CHANGELOG.md:13]: Ok for me -- I may rewrite it when I implement the version we discussed.

**Review by odelalleau — APPROVED** [review]

**Pull request merged**